### PR TITLE
Feature.ohm 175 extract imathas

### PR DESCRIPTION
**Pull request by sketchings**

### CODE REVIEW

File: admin/approvepending2.php [hunk 1]
```diff
@@ -228,7 +229,7 @@ function getGroups() {
 	$curBreadcrumb .= "&gt; <a href=\"../util/utils.php\">Utilities</a> &gt; ";
 }
 
-require("../header.php");
+require(IMATHAS_DIR . "/header.php");
```
Comment (drlippman) [inline admin/approvepending2.php]: Where is IMATHAS_DIR defined?

File: directaccess.php [hunk 2]
```diff
@@ -261,9 +263,9 @@ function setlogintype(n) {
...
-<span class=form><label for="firstname">Enter First Name:</label></span> <input class=form type=text size=20 id=firstname name=firstname autocomplete="given-name" <?php if (isset($_POST['firstname'])) { printf('value="%s"', Sanitize::encodeStringForDisplay($_POST['firstname'])); } ?>><BR class=form>
-<span class=form><label for="lastname">Enter Last Name:</label></span> <input class=form type=text size=20 id=lastname name=lastname autocomplete="family-name" <?php if (isset($_POST['lastname'])) { printf('value="%s"', Sanitize::encodeStringForDisplay($_POST['lastname'])); } ?>><BR class=form>
-<span class=form><label for="email">Enter E-mail address:</label></span>  <input class=form type=text size=60 id=email name=email autocomplete="email" <?php if (isset($_POST['email'])) { printf('value="%s"', Sanitize::encodeStringForDisplay($_POST['email'])); } ?>><BR class=form>
+<span class=form><label for="firstname">Enter First Name:</label></span> <input class=form type=text size=20 id=firstname name=firstname <?php if (isset($_POST['firstname'])) { printf('value="%s"', Sanitize::encodeStringForDisplay($_POST['firstname'])); } ?>><BR class=form>
```
Comment (drlippman) [inline directaccess.php]: the autocomplete fields should be included (it's part of one of the a11y guidelines)

File: includes/calendardisp.php [hunk 3]
```diff
@@ -90,7 +90,7 @@ function showcalendar($refpage) {
 	echo '>'.$i.'</option>';
 }
 echo '</select> weeks. ';
-echo '<a href="#" onclick="hidevisualcal();return false;" title="'._('Hide visual calendar and display events list').'" aria-controls="caleventslist">';
+echo '<a href="#" onclick="hidevisualcal();return false;" aria-label="'._('Hide visual calendar and display events list').'" title="'._('Hide visual calendar and display events list').'" aria-controls="caleventslist">';
```
Comment (drlippman) [inline includes/calendardisp.php]: aria-label should be removed to avoid inconsistency with visual label

File: index.php [hunk 4]
```diff
@@ -522,18 +547,14 @@ function printCourses($data,$title,$type=null,$hashiddencourses=false) {
...
-		echo '<br/><a class="small" href="admin/forms.php?action=findstudent&from=home">',_('Find Student'),'</a> ';
+		echo '<a class="small" href="admin/modcourseorder.php?type='.$type.'">Change Course Order</a>';
 	}
+	echo '</div><div class="center">';
+	echo '<a id="unhidelink'.$type.'" '.($hashiddencourses?'':'style="display:none"').' class="small" href="admin/unhidefromcourselist.php?type='.$type.'">View hidden courses</a>';
 	echo '</div>';
 	if ($type=='teach' && ($myrights>=75 || ($myspecialrights&4)==4)) {
 		echo '<div class="center"><a class="abutton" href="admin/admin2.php">', _('Admin Page'), '</a></div>';
 	}
-
```
Comment (drlippman) [inline index.php]: Revert this block (525-536) back to the IMathAS code, which has added a feature that just hadn't made it to MOM yet.

File: admin/approvepending2.php (hunk 1 above)
Comment (sketchings) [reply admin/approvepending2.php]: not used until mom

File: index.php (hunk 4 above)
Comment (sketchings) [reply index.php:536]: I THINK I set this like you wanted. Let me know

File: directaccess.php (hunk 2 above)
Comment (sketchings) [reply directaccess.php]: no longer an issue

**Comment by drlippman** [comment]
Merged manually.

**Pull request closed without merging**